pkg/manifests/manifests.go

```diff
@@ -1425,6 +1429,10 @@ func (f *Factory) PrometheusK8s(grpcTLS *v1.Secret, trustedCABundleCM *v1.Config
 		p.Spec.Retention = f.config.ClusterMonitoringConfiguration.PrometheusK8sConfig.Retention
 	}
 
+	if !isUnsetRetentionConfig(f.config.ClusterMonitoringConfiguration.PrometheusK8sConfig.RetentionSize) {
```

Comment:
I feel that `if !isUnsetRetentionConfig(...)` can be confusing to read with the double negation ("unless retention is not set"). I'd be more inclined to remove the intermediate function.

```suggestion
	if f.config.ClusterMonitoringConfiguration.PrometheusK8sConfig.RetentionSize != "" && f.config.ClusterMonitoringConfiguration.PrometheusK8sConfig.RetentionSize != "0" {
```